File: FinalProject/app/src/main/java/com/example/finalproject/items/Weapon.java
package com.example.finalproject.items;

public class Weapon extends Item {
    private int damage, typeOfDamage, typeOfWeapon;

    public Weapon(int damage, int typeOfDamage, int typeOfWeapon, String name) {
        super.setName(name);
        this.damage = damage;
        this.typeOfDamage = typeOfDamage;
        this.typeOfWeapon = typeOfWeapon;
    }

    public int getDamage() {
        return damage;
    }

    public void setDamage(int damage) {
        this.damage = damage;
    }

    public int getTypeOfDamage() {
        return typeOfDamage;
    }

    public void setTypeOfDamage(int typeOfDamage) {
        this.typeOfDamage = typeOfDamage;
    }

    public int getTypeOfWeapon() {
        return typeOfWeapon;
    }

    public void setTypeOfWeapon(int typeOfWeapon) {
        this.typeOfWeapon = typeOfWeapon;
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/items/Weapon.kt
package com.example.finalproject.items

class Weapon(val damage:Double, val typeOfDamage:Int, val typeOfWeapon:Int,
             costSell:Double, costBuy:Double, rarity:Int, category:Int, name:String)
    :Item(costSell, costBuy, rarity, category, name){
        constructor(weapon: Weapon):this(weapon.damage, weapon.typeOfDamage, weapon.typeOfWeapon,
            weapon.costSell, weapon.costBuy,
            weapon.rarity, weapon.category, weapon.name)
    }